src/test/java/BoardTests.java

```diff
@@ -98,6 +128,20 @@ void testHasBoardMoreThanOneFreeCell() {
         Assertions.assertTrue(board.hasBoardMoreThanOneFreeCell());
     }
 
+    @ParameterizedTest
+    @MethodSource("provideAdjacentBoardPositions")
+    void testGetAdjacentCells(Position position, Set<Position> adjacentPositions, Class<Exception> expectedException) {
+        if (expectedException != null) {
+            assertThrows(expectedException, () -> board.getAdjacentCells(position));
+        } else {
+            Set<Cell> adjacentCells = new HashSet<>(4);
```

Comment:
see the above comment about the number of adjacent cells

Reply:
Corrected